src/Phinx/Config/Config.php

```diff
@@ -153,6 +153,49 @@ public function getEnvironment($name)
                     $this->values['environments']['default_migration_table'];
             }
 
+            if (!empty($environments[$name]['name']) && $database !== null) {
+                if ($database != $environments[$name]['name']) {
+                    throw new \InvalidArgumentException(sprintf(
+                        'The database "%s" does not exists in environment "%s"',
+                        $database,
+                        $name
+                    ));
+                }
+                $environments[$name]['name'] = $database;
+            }
+            elseif (!empty($environments[$name]['databases']) && $database !== null ) {
```

Comment:
Moslt likely not. Proper value is verified in next condition.
`$database !== null` is in condition to verify if parameter was sent, or default value is provided.
Imho no need to check if `$database === false` but if you want, i can add it.